src/main/webapp/app/exercises/shared/plagiarism/plagiarism-split-view/split-pane-header/split-pane-header.component.ts

```diff
@@ -14,16 +16,87 @@ export type FileWithHasMatch = {
     templateUrl: './split-pane-header.component.html',
     styleUrls: ['./split-pane-header.component.scss'],
 })
-export class SplitPaneHeaderComponent implements OnChanges {
+export class SplitPaneHeaderComponent implements OnChanges, OnInit, OnDestroy {
     @Input() files: FileWithHasMatch[];
     @Input() studentLogin: string;
+    @Input() fileSelectedSubject!: Subject<TextPlagiarismFileElement>;
+    @Input() isLockFilesEnabled!: boolean;
+    @Input() showFilesSubject!: Subject<boolean>;
+    @Input() dropdownHoverSubject!: Subject<TextPlagiarismFileElement>;
```

Comment:
Can you change this to use the new client coding guidelines, using signals? And can you turn the component into standalone?

Reply:
Adapted to use input signal instead of @Input directive and adapted tests. 

Turning the component into a standalone is not possible without affecting unrelated parts of the issue due to the parent component also needing to be a standalone, leading to cascading refactors. Since currently migration is being done, I believe it is not a good idea to touch unrelated code. I am responsible for refactoring of this module and will address it as part of migration.